test/e2e/standard/cluster.go

```diff
@@ -400,3 +401,32 @@ func (sc *SanityChecker) checkCanUseAzureFileStorage(ctx context.Context) error
 
 	return nil
 }
+
+func (sc *SanityChecker) checkCantDoDockerBuild(ctx context.Context) error {
+	namespace, err := sc.createProject(ctx)
+	if err != nil {
+		return err
+	}
+	defer sc.deleteProject(ctx, namespace)
+
+	_, err = sc.Client.EndUser.BuildV1.BuildConfigs(namespace).Create(&buildv1.BuildConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "test",
+		},
+		Spec: buildv1.BuildConfigSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{
+					Dockerfile: to.StringPtr("FROM scratch"),
+				},
+				Strategy: buildv1.BuildStrategy{
+					Type: buildv1.DockerBuildStrategyType,
+				},
+			},
+		},
+	})
+	if !kerrors.IsForbidden(err) {
```

Comment:
a better message would be `if err == nil return fmt.Error("cluster can docker build")` before checking if the error is checked vs kerrors, otherwise you would get "unexpected error nil" which can be confusing

Reply:
I'm comfortable with this as it is, even if parsimonious